src/posts/2020-12-21-ebpf-updates/index.md

```diff
@@ -0,0 +1,393 @@
+  variables may present a security issue. This short post relies on bcc's
+  execsnoop tool to show that MySQL passwords can be retrieved by examining the
+  arguments passed to the system call, on the kernel side.
+
+* [_Why We Switched from bcc-tools to libbpf-tools for BPF Performance Analysis_](https://en.pingcap.com/blog/why-we-switched-from-bcc-tools-to-libbpf-tools-for-bpf-performance-analysis),
+  from Wenbo Zhang.  
+  Another article on the benefits brought by CO-RE, for which libbpf provides
+  good support. After comparing bcc-based and libbpf-based tracing tools in
+  terms of features and memory footprint, the author provide a list of tools
+  and invocation patterns they use to analyze I/O performance.
+
+* [_Tips and Tricks for Writing Linux BPF Applications with libbpf_](https://pingcap.com/blog/tips-and-tricks-for-writing-linux-bpf-applications-with-libbpf),
+  from Wenbo Zhang.  
+  This is a collection of hints about libbpf, to make it easier to write
+  applications or to get better performance. The article contains several hints
+  related to the use of eBPF “skeletons” generated by bpftool, eBPF maps, eBPF
+  global variables, or pointers dereferencing.
+
+* [_BPF and Go: Modern forms of introspection in Linux_](https://medium.com/bumble-tech/bpf-and-go-modern-forms-of-introspection-in-linux-6b9802682223),
+  from Marko Kevac.  
+  This long read contains two parts. The first one is a good introduction, in
+  terms of context and history, to eBPF. The second part focuses on the
+  interactions between eBPF. If it is not possible to write eBPF programs in Go
+  at this date, several Go libraries are available to write user applications
+  to manage eBPF objects in this language. As for observability, Go has some
+  particularities that make some it harder or impossible to get thread id for
+  goroutines or use uretprobes. But the article demonstrates that a large part
+  of eBPF's tracing capabilities remain available for Go applications.
+
+* [_Adding BPF target support to the Rust compiler_](https://confused.ai/posts/rust-bpf-target),
+  from Alessandro Decina.  
+  eBPF bytecode can be compiled from C with clang/LLVM's backend. Other
+  languages, and Rust in particular, use the same toolchain. Compiling Rust
+  code into eBPF is not straightforward, but some hacks permit it. This post
+  provides details on the linking steps, where the issue lies, and introduces a
+  new linker and two related compiler targets for rustc, to ease compilation
+  from Rust to eBPF bytecode. Although the targets are not merged yet, a
+  command-line hack allows developers to use it with the stable versions of
+  rustc and cargo. The linker is available [on
+  GitHub](https://github.com/alessandrod/bpf-linker).
+
+### Academic Research
+
+* [_Express Data Path Kernel Objects for Real-Time Audio Streaming Optimization_](https://lac2020.sciencesconf.org/data/proceedings.pdf#section*.12),
```

Comment:
This is quite a big shootout for a paper to a non-international conference. Could we at least clarify the conference in the first line (`Linux Audio Conference 2020`)?

Reply:
Can do, or do you think I should simply drop the subsection and move it above with the other presentations maybe?

Reply:
Yeah, that would also be a solution IMO. I tend to hold academic papers to higher/different standards than blog posts, and in particular, I wouldn't want us to put e.g. a good OSDI paper and a non-international workshop paper on the same level.